DigitalLearningSolutions.Web.AutomatedUiTests/AccessibilityTests.cs

```diff
@@ -0,0 +1,50 @@
+namespace DigitalLearningSolutions.Web.AutomatedUiTests
+{
+    using System;
+    using FluentAssertions;
+    using OpenQA.Selenium;
+    using OpenQA.Selenium.Chrome;
+    using Selenium.Axe;
+    using Xunit;
+
+    public class AccessibilityTests : IClassFixture<SeleniumServerFactory<Startup>>, IDisposable
+    {
+        private readonly string baseUrl;
+        private readonly IWebDriver driver;
+
+        public AccessibilityTests(SeleniumServerFactory<Startup> factory)
+        {
+            factory.CreateClient();
+            baseUrl = factory.RootUri;
+            driver = CreateHeadlessChromeDriver();
+        }
+
+        public void Dispose()
+        {
+            driver.Quit();
+            driver.Dispose();
+        }
+
+        [Theory]
+        [InlineData("/Home")]
+        [InlineData("/Login")]
+        [InlineData("/ForgotPassword")]
+        [InlineData("/Register")]
+        public void Page_has_no_accessibility_errors(string url)
+        {
+            // when
+            driver.Navigate().GoToUrl(baseUrl + url);
```

Comment:
As this goes directly to a url, does this mean it won't test pages that can't be reached through regular GET requests? e.g. the Choose A Centre page when logging in?

Reply:
It would be nice if the controller actions could be called instead of just heading to a url, but then I am very unfamiliar with this and have no idea if that's possible. Either way, I imagine it won't matter for the vast majority of pages, so it will just be a case of not relying on the tests when making those specific pages, which we've done up to this point anyway. So I'm happy with this.

Reply:
The Selenium tests control the browser - they effectively open Chrome locally and interact with it. Just like you cannot select "POST" in normal Chrome browser when you enter a URL, you won't be able to do it when navigating in these Selenium tests (and it would defeat the point as we're trying to mimic user's experience). However (!), how the tests do work is that they allow you to interact with the browser, so you can fill in forms and press buttons (e.g. log in), which will allow us to test these pages. Given that the accessibility scanner is a bit slow, I would prefer to keep these tests limited (rather than accessibility testing every version of the page), but we can definitely discuss it if there are pages that would strongly benefit from several scans. 😄